cpmpy/solvers/scip.py

```
@@ -0,0 +1,419 @@
+        self.scip_model.setParams(kwargs)
+
+        _ = self.scip_model.optimize()
+        scip_objective = self.scip_model.getObjective()
+
+        scip_status = self.scip_model.getStatus()
+
+        # new status, translate runtime
+        self.cpm_status = SolverStatus(self.name)
+        self.cpm_status.runtime = self.scip_model.getSolvingTime()
+
+        # translate exit status
+        if scip_status == "optimal":
+            if self.has_objective():
+                self.cpm_status.exitstatus = ExitStatus.OPTIMAL
+            else:
+                self.cpm_status.exitstatus = ExitStatus.FEASIBLE
+        elif scip_status == "infeasible":
+            self.cpm_status.exitstatus = ExitStatus.UNSATISFIABLE
+        elif scip_status.endswith("limit"):  # timelimit, nodelimit, ...
+            self.cpm_status.exitstatus = ExitStatus.FEASIBLE
+        elif scip_status == "unknown":
+            self.cpm_status.exitstatus = ExitStatus.UNKNOWN
+        # TODO: "userinterrupt" and check if there is a solution?
+        else: # there are many more, e.g. one for each limit
+            raise NotImplementedError(
+                f"Translation of scip status {scip_status} to CPMpy status not implemented")  # a non-mapped status type, please report on github
+
+        # True/False depending on self.cpm_status
+        has_sol = self._solve_return(self.cpm_status)
+
+        # translate solution values (of user specified variables only)
+        self.objective_value_ = None
+        if has_sol:
+            # fill in variable values
+            for cpm_var in self.user_vars:
+                solver_val = self.scip_model.getVal(self.solver_var(cpm_var))
+                if cpm_var.is_bool(): 
+                    cpm_var._value = solver_val >= 0.5
+                else:
+                    cpm_var._value = int(solver_val)
+            # set _objective_value
+            if self.has_objective():
+                self.objective_value_ = scip_objective.getObjVal()
```

This should raise an error. The ```getObjVal()``` function can only be called for ```scip_model```, not for the objective expression. So I'd say that you can also remove the line: 
```scip_objective = self.scip_model.getObjective()```

---

It did : ) Fixed it and will push it soon